tests/vector_field_nets_test.py

```diff
@@ -0,0 +1,217 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+from __future__ import annotations
+
+from typing import Tuple
+
+import pytest
+import torch
+import torch.nn as nn
+
+from sbi.neural_nets.embedding_nets import CNNEmbedding
+from sbi.neural_nets.net_builders.vector_field_nets import (
+    VectorFieldMLP,
+    VectorFieldTransformer,
+    build_adamlp_network,
+    build_standard_mlp_network,
+    build_transformer_network,
+)
+
+
+@pytest.mark.parametrize("input_event_shape", ((1,), (4,)))
+@pytest.mark.parametrize("condition_event_shape", ((1,), (7,)))
+@pytest.mark.parametrize("batch_dim", (1, 10))
```

Comment:
Should we also test with multiple batch dimensions?

Reply:
Multiple batch dimensions are fine, but all nets will only have to work with shape (b, ...) i.e. with a single batch dimension as the estimators will anyway flatten it.

Reply:
So I did tes ta few batch_shapes now, nut for combinatorial reasons it doesnt make sense to test more (this already runs a while just for shape checking). Ended up reducing some of the hidden_dims/batch_dims and so on to speed it up. 

As said in the previous comment. All nets only need to support (b, event) and (b, cond). All the reshaping logic is in the estimators.